scipy/sparse/tests/test_coo.py

```diff
@@ -102,43 +164,98 @@ def test_reshape():
     assert row_vec.shape == (1, 3)
     assert_equal(row_vec.toarray(), np.array([[1, 0, 3]]))
 
+    # attempting invalid reshape
+    with pytest.raises(ValueError, match="cannot reshape array"):
+        arr1d.reshape((3,3))
+    
+    # attempting reshape with a size 0 dimension
+    with pytest.raises(ValueError, match="cannot reshape array"):
+        arr1d.reshape((3,0))
+    
+
+def test_reshape_2d():
+    # reshaping 2d sparse arrays
     arr2d = coo_array([[1, 2, 0], [0, 0, 3]])
     assert arr2d.shape == (2, 3)
-
-    flat = arr2d.reshape((6,))
-    assert flat.shape == (6,)
-    assert_equal(flat.toarray(), np.array([1, 2, 0, 0, 0, 3]))
-
-
-def test_nnz():
-    arr1d = coo_array([1, 0, 3])
-    assert arr1d.shape == (3,)
-    assert arr1d.nnz == 2
-
-    arr2d = coo_array([[1, 2, 0], [0, 0, 3]])
-    assert arr2d.shape == (2, 3)
-    assert arr2d.nnz == 3
-
-
-def test_transpose():
-    arr1d = coo_array([1, 0, 3]).T
-    assert arr1d.shape == (3,)
-    assert_equal(arr1d.toarray(), np.array([1, 0, 3]))
-
-    arr2d = coo_array([[1, 2, 0], [0, 0, 3]]).T
-    assert arr2d.shape == (3, 2)
-    assert_equal(arr2d.toarray(), np.array([[1, 0], [2, 0], [0, 3]]))
-
-
-def test_transpose_with_axis():
-    arr1d = coo_array([1, 0, 3]).transpose(axes=(0,))
-    assert arr1d.shape == (3,)
-    assert_equal(arr1d.toarray(), np.array([1, 0, 3]))
-
-    arr2d = coo_array([[1, 2, 0], [0, 0, 3]]).transpose(axes=(0, 1))
-    assert arr2d.shape == (2, 3)
-    assert_equal(arr2d.toarray(), np.array([[1, 2, 0], [0, 0, 3]]))
-
+    
+    # 2d to 2d
+    to2darr = arr2d.reshape((3,2))
+    assert to2darr.shape == (3,2)
+    assert_equal(to2darr.toarray(), np.array([[1, 2], [0, 0], [0, 3]]))
+    
+    # 2d to 1d
+    to1darr = arr2d.reshape((6,))
+    assert to1darr.shape == (6,)
+    assert_equal(to1darr.toarray(), np.array([1, 2, 0, 0, 0, 3]))
+    
+    # 2d to 3d
+    to3darr = arr2d.reshape((2, 3, 1))
+    assert to3darr.shape == (2, 3, 1)
+    assert_equal(to3darr.toarray(), np.array([[[1], [2], [0]], [[0], [0], [3]]]))
+
+    # attempting invalid reshape
+    with pytest.raises(ValueError, match="cannot reshape array"):
+        arr2d.reshape((1,3))
+
+
+@pytest.mark.parametrize(('shape', 'new_shape'), [((4,9,6,5), (3,6,15,4)), ((4,9,6,5),
+                                            (36,30)), ((4,9,6,5), (1080,)), ((4,9,6,5),
+                                                                    (2,3,2,2,3,5,3))])
```

Comment:
```suggestion
@pytest.mark.parametrize(('shape', 'new_shape'), [((4,9,6,5), (3,6,15,4)),
                                                  ((4,9,6,5), (36,30)),
                                                  ((4,9,6,5), (1080,)),
                                                  ((4,9,6,5), (2,3,2,2,3,5,3)),])
```

(the trailing `,` is conventional, so that the line does not need to be changed to add to the list in future)